src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -603,37 +626,13 @@ internal bool IsEntryPointCandidate
         /// </summary>
         internal bool HasEntryPointSignature(CSharpCompilation compilation)
         {
-
-            bool IsAsyncMainReturnType(TypeSymbol type)
-            {
-                var namedType = type as NamedTypeSymbol;
-                if (namedType == null)
-                {
-                    return false;
-                }
-                else if (namedType.ConstructedFrom == compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_Task))
-                {
-                    // Change this to `namedType.IsNonGenericTaskType` if you want to support "task-like" objects.
-                    return true;
-                }
-                else if (namedType.ConstructedFrom == compilation.GetWellKnownType(WellKnownType.System_Threading_Tasks_Task_T))
-                {
-                    // Change this to `namedType.IsGenericTaskType` if you want to support "task-like" objects.
-                    return namedType.TypeArguments[0].SpecialType == SpecialType.System_Int32;
-                }
-                else
-                {
-                    return false;
-                }
-            }
-
             if (IsVararg)
             {
                 return false;
             }
 
             TypeSymbol returnType = ReturnType;
-            bool isAsyncMainReturnType = IsAsyncMainReturnType(returnType);
+            bool isAsyncMainReturnType = HasAsyncMainReturnType(compilation);
```

Comment:
>isAsyncMainReturnType [](start = 17, length = 21)

The name of the local is confusing, it implies that the method is async, yet it may disagree with ```IsAsync``` value.